contributors/design-proposals/federated-deployment.md

```diff
@@ -0,0 +1,127 @@
+# Federated Deployment
+# Design Document
+
+Authors: [Marcin Wielgus](mailto:mwielgus@google.com), [Nikhil Jindal](mailto:nikhiljindal@google.com )
+
+## Introduction
+
+The purpose of this document is to provide a detailed design of how Kubernetes Deployments should be handled in Federation, with the special emphasis on rolling update support in deployment updates. 
+
+The internal design is based on [Federated ReplicaSet](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/federated-replicasets.md)
+
+## Background
+
+With Federation we are trying to provide Kubernetes objects and functionality spanning over multiple clusters. One of the key objects in Kubernetes is Deployment. Deployment allows users to maintain a defined number of pod replicas and conveniently update them if needed. The update functionality is the main thing that differentiate them from replica sets. ReplicaSets update everything at once while Deployments allow a slow, rolling update and rollback in case the update is unsuccessful. 
+
+## Requirements
+
+Federated deployments must provide the similar functionality as regular deployments, in particular :
+  + [R1] Work well with kubectl rollout
+    + History - provide all of the revisions of federated deployment.
+    + Pause/Resume - provide an ability to pause and resume an ongoing update
+    + Status - provide rollout status information
+    + Undo - provide go back to the previous version
+    + Require no/almost-no fixes in kubectl
```

Comment:
No. same kubectl commads (like `kubectl get rs`) can work for both kube-apiserver and federation-apiserver.